document/js-api/index.bs

```diff
@@ -1115,6 +1147,111 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 
 </div>
 
+<h3 id="jspi">JavaScript Promise Integration</h3>
+
+A {{Suspending}} object represents a JavaScript function whose calls should be *suspended* when they return a Promise object.
+Each {{Suspending}} object has a \[[wrappedFunction]] internal slot which holds a JavaScript function.
+
+In addition, the {{promising}} function takes as argument a WebAssembly function and returns a JavaScript function that returns a Promise that is resolved when the WebAssembly function completes.
+
+<pre class="idl">
+[Exposed=*]
+partial namespace WebAssembly {
+    Function promising(Function wasmFun);
+};
+
+[LegacyNamespace=WebAssembly, Exposed=*]
+interface Suspending {
+    constructor(Function jsFun);
+};
+</pre>
+
+<div algorithm>
+    The  <dfn method for="WebAssembly">promising(|wasmFun|)</dfn> function, when invoked, performs the following steps:
+    1. If [$IsCallable$](|wasmFun|) is false, throw a {{TypeError}}.
+    1. If |wasmFun| does not have a \[[FunctionAddress]] internal slot, throw a {{TypeError}}.
+    1. Let |builder| be a new [=AbstractClosure=] with parameters that captures |wasmFun| that, when invoked, performs [=run a Promising function=].
```

Comment:
Do you need to give the parameters a name to pass them along to `[=run a Promising function=]`?